R/rows-dm.R

```diff
@@ -181,21 +182,21 @@ dm_rows_delete <- function(x, y, ..., in_place = NULL) {
 #'
 #' @rdname rows-dm
 #' @export
-dm_rows_truncate <- function(x, y, ..., in_place = NULL) {
+dm_rows_truncate <- function(x, y, ..., in_place = NULL, progress = NA) {
   check_dots_empty()
 
-  dm_rows(x, y, rows_truncate, top_down = FALSE, in_place, require_keys = FALSE)
+  dm_rows(x, y, rows_truncate, top_down = FALSE, in_place, require_keys = FALSE, progress = progress)
 }
 
-dm_rows <- function(x, y, operation, top_down, in_place, require_keys) {
+dm_rows <- function(x, y, operation, top_down, in_place, require_keys, progress = NA) {
```

Comment:
OTOH, we might need a lookup from e.g. `"insert"` to `dm_rows_insert()` later:

```suggestion
dm_rows <- function(x, y, operation_name, top_down, in_place, require_keys, progress = NA) {
```

It can be implemented in a discoverable way:

```r
get_dm_rows_op <- function(opreration_name) {
  switch(operation_name,
    "insert" = dm_rows_insert,
    ...
  )
}
```

Reply:
done in https://github.com/cynkra/dm/pull/573/commits/26ee53f7eb43c1620312c0820daad542ac015aa2